Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py

```diff
@@ -0,0 +1,193 @@
+                'path': file_result['FileID'],
+                'name': attachment.get('filename')
+            })
+        return file_names
+
+    def convert_to_incident(self) -> Dict[str, Any]:
+        """
+        Convert an Email class instance to a demisto incident
+        Returns:
+            A dict with all relevant fields for an incident
+        """
+        return {
+            'occurred': self.date.isoformat(),
+            'created': datetime.now(timezone.utc).isoformat(),
+            'details': self.text or self.html,
+            'name': self.subject,
+            'attachment': self.parse_attachment(),
+            'rawJSON': json.dumps(self.raw_json)
+        }
+
+    def generate_raw_json(self):
+        raw_json = {
+            'to': self.to,
+            'cc': self.cc,
+            'from': self.from_,
+            'format': self.format,
+            'text': self.text,
+            'subject': self.subject,
+            'attachments': ','.join([attachment['filename'] for attachment in self.attachments]),
+            'rawHeaders': self.headers
+        }
+        if self.html:
+            raw_json['html'] = self.html
+        if self.raw_body:
+            raw_json['rawBody'] = self.raw_body
+        return raw_json
+
+
+def fetch_incidents(client,
+                    last_run,
+                    first_fetch_time):
+    """
+    This function will execute each interval (default is 1 minute).
+
+    Args:
+        client (Client): HelloWorld client
+        last_run (dateparser.time): The greatest incident created_time we fetched from last fetch
+        first_fetch_time (dateparser.time): If last_run is None then fetch all incidents since first_fetch_time
+
+    Returns:
+        next_run: This will be last_run in the next fetch-incidents
+        incidents: Incidents that will be created in Demisto
+    """
+    # Get the last fetch time, if exists
+    last_fetch = last_run.get('last_fetch')
+
+    # Handle first time fetch
+    if last_fetch is None:
+        latest_created_time = dateparser.parse(f'{first_fetch_time} UTC')
+    else:
+        latest_created_time = datetime.fromisoformat(last_fetch)
+    messages_query = generate_search_query(latest_created_time, PERMITTED_FROM_ADDRESSES, PERMITTED_FROM_DOMAINS)
+    messages = client.search(messages_query)
+    messages = messages[:50]
+    mails_fetched = []
+    for uid, message_data in client.fetch(messages, 'RFC822').items():
+        message_bytes = message_data.get(b'RFC822')
+        if not message_bytes:
+            continue
+        email_message = mailparser.parse_from_bytes(message_bytes)
+        # The search query filters emails by day, not by exact date
+        email_message_object = Email(email_message)
+        if email_message_object.date > latest_created_time:
+            mails_fetched.append(email_message_object)
+    if mails_fetched:
+        latest_created_time = max(mails_fetched, key=lambda x: x.date).date
```

Comment:
feels to me like another redundant loop, we can do it in the loop in which we build `mails_fetched`

Reply:
We could but that will uglify the code because it will have to happen under the `if email_message_object.date > latest_created_time`.

As long as we have a limit the loop should be pretty fast and we get a much readable code IMHO.

Reply:
ok your call